x-pack/plugins/security_solution/server/lib/detection_engine/schemas/rule_converters.test.ts

```diff
@@ -206,27 +206,42 @@ describe('rule_converters', () => {
   });
 
   describe('convertPatchAPIToInternalSchema', () => {
-    test('should not update version for immutable rules', () => {
-      const patchParams = {
+    test('should set version to one specified in next params', () => {
+      const nextParams = {
+        index: ['new-test-index'],
+        language: 'lucene',
+        version: 3,
+      };
+      const existingRule = getRuleMock({ ...getQueryRuleParams() });
+      const patchedParams = convertPatchAPIToInternalSchema(nextParams, existingRule);
+      expect(patchedParams).toEqual(
+        expect.objectContaining({
+          params: expect.objectContaining({ version: 3 }),
+        })
+      );
+    });
+
+    test('should not update version for immutable rules if it is not specified in next params', () => {
+      const nextParams = {
         index: ['new-test-index'],
         language: 'lucene',
       };
-      const rule = getRuleMock({ ...getQueryRuleParams(), immutable: true });
-      const patchedParams = convertPatchAPIToInternalSchema(patchParams, rule);
+      const existingRule = getRuleMock({ ...getQueryRuleParams(), immutable: true });
```

Comment:
We could explicitly set the version to some number here too (e.g. `42`) and then assert that `convertPatchAPIToInternalSchema` doesn't change it. That would decouple the test from the rule mock it uses. Otherwise, if someone changes the `version` in the mock, it will break this test.

Same for other tests in this file which test the version update logic.

Reply:
Good suggestion, thanks 👍